### DISCUSSION TITLE
Can't use workflow outputs

### DISCUSSION THREAD

**Discussion by tdudgeon** [nextflow-io/nextflow#2722] [Q&A] [answered]
I've asked this on gitter with no responses, so trying there.

I'm trying to work out how to get almost real time feedback as a workflow executes.
I'm using DSL2 with a workflow that follows the split-process-join pattern. The process part runs as multiple NF processes in parallel, and its those that I'm wanting to monitor, most likely by writing a line to a common file that records how many units of work have been performed when each NF process completes. Generating the units of work is simple. It's how that get's written to the common file in a safe manner that I'm not sure about. I have the NF process writing the units of work to an output channel. I was thinking I need to subscribe to that channel in the main workflow an write to the file. Something like this:

```
workflow my_workflow {

    take:
    input1

    main:
    split(input1)
    my_process(split.out.flatten())
    collect_results(my_process.out[0].collect())

    emit:
    collect_results.out
    my_process.out[1]
}

workflow {
    my_workflow(input_file)
}

File file = new File("out.txt")
my_workflow.out[1].subscribe {
    file.write "$it\n"
}
```

But that's not working. I get this error:
`Access to 'my_workflow.out' is undefined since workflow doesn't declare any output.`

The [docs](https://www.nextflow.io/docs/latest/dsl2.html#workflow-outputs) specifically state that:
`the result of the my_pipeline execution can be accessed using the out property ie. my_pipeline.out.`

**Comment by jorgeaguileraseqera** [comment, marked as answer]
not sure if solved it but did you try to include the subscribe into the workflow closure?

```
workflow {
    my_workflow(input_file)
    File file = new File("out.txt")
    my_workflow.out[1].subscribe {
         file.write "$it\n"
    }
}
```

**Reply by drpatelh** [reply]
I was going to suggest the same thing 💯

**Reply by drpatelh** [reply]
`my_workflow.out` isn't available outside the invoking `workflow` scope which is why you would have been getting this error:

```
Access to 'my_workflow.out' is undefined since workflow doesn't declare any output.
```

**Comment by tdudgeon** [comment]
@jorgeaguileraseqera Thanks. That did the trick!
